app/grandchallenge/workstations/views.py

```diff
@@ -301,10 +302,10 @@ def form_valid(self, form):
         )
 
         url = session.get_absolute_url()
-
+        url += f"?path={quote_plus(self.kwargs.get('workstation_path', ''))}"
         qs = self.request.META.get("QUERY_STRING", "")
         if qs:
-            url = f"{url}?{qs}"
+            url = f"{url}&qs={quote_plus(qs)}"
 
         return HttpResponseRedirect(url)
```

Comment:
The redirect here needs to load SessionDetail so it cannot have the cirrus path added to it because that would redirect to the session_proxy instead of SessionDetail. Therefore, the cirrus path is sent through as a query parameter. This ensures the correct view is loaded. In the SessionDetail template, the query parameters are parsed and converted back into the correct URL to load the workstation.